client/src/elm/Pages/Utils.elm

```diff
@@ -1246,3 +1250,38 @@ customPopup language showWarning actionLabel ( topMessage, bottomMessage, action
 unique : List a -> List a
 unique =
     EverySet.fromList >> EverySet.toList
+
+
+viewBySyncStatus : Language -> HealthCenterId -> SyncManager.Model.SyncInfoAuthorityZipper -> Html msg -> Html msg
+viewBySyncStatus language healthCenterId syncInfoAuthorities contentForView =
+    let
+        selectedHealthCenterSyncInfo =
+            syncInfoAuthorities
+                |> Maybe.andThen
+                    (Zipper.toList >> List.Extra.find (\authorityInfo -> authorityInfo.uuid == fromEntityUuid healthCenterId))
+
+        showWarningMessage header message =
+            div [ class "ui message warning" ]
+                [ div [ class "header" ] [ text <| translate language header ]
+                , text <| translate language message
+                ]
+    in
+    selectedHealthCenterSyncInfo
+        |> Maybe.map
+            (\syncInfo ->
+                case syncInfo.status of
+                    SyncManager.Model.NotAvailable ->
+                        showWarningMessage Translate.SelectedHCNotSynced Translate.PleaseSync
+
+                    SyncManager.Model.Downloading ->
+                        if syncInfo.remainingToDownload > 1500 then
```

Comment:
@anvmn What is this 1500? Seconds? Size? any reason for this particular number?

Reply:
Added comment